eodatasets/prepare/modis_usgs_mcd43a1.py

```diff
@@ -0,0 +1,228 @@
+    Separate references are provided for each of the brdf parameter bands:
+        volumetric (vol), isometric (iso) and geometric (geo)
+
+    """
+    band_info = {
+        'bands': {},
+    }
+    with rasterio.open(imagery_file, 'r') as collection:
+        for ds in collection.subdatasets:
+            raster_params = re.match('(?P<fmt>HDF4_EOS:EOS_GRID):(?P<path>[^:]+):(?P<layer>.*)$', ds)
+            if '_Quality_' in raster_params['layer']:
+                name = raster_params['layer'].split(':')[1]
+                band_info['bands'][name] = {
+                    'path': Path(raster_params['path']).name,
+                    'layer': raster_params['layer']
+                }
+            else:
+                name = raster_params['layer'].split(':')[1]
+                # BRDF parameter bands are isotropic, volumetric and geometric
+                for idx, band_name in enumerate(['iso', 'vol', 'geo'], 1):
+                    band_info['bands'][name + '_' + band_name] = {
+                        'path': Path(raster_params['path']).name,
+                        'layer': raster_params['layer'],
+                        'band': idx
+                    }
+    return band_info
+
+
+def get_bounds(polygon):
+    """
+    Returns the bounds of the polygon
+    """
+    min_lon, min_lat, max_lon, max_lat = \
+        polygon.bounds
+
+    return {
+        'll': {
+            'lat': min_lat,
+            'lon': min_lon,
+        },
+        'lr': {
+            'lat': min_lat,
+            'lon': max_lon,
+        },
+        'ur': {
+            'lat': max_lat,
+            'lon': max_lon,
+        },
+        'ul': {
+            'lat': max_lat,
+            'lon': min_lon,
+        }
+    }
+
+
+def rio_spatial_projection(src_dataset: str):
+    """
+    Returns the spatial projection from a rasterio readable dataset
+    """
+    with rasterio.open(src_dataset, 'r') as fd:
+        return _spatial_projection(fd.shape, fd.transform, fd.crs.wkt)
+
+
+def _spatial_projection(shape, transform, spatial_reference):
+    """
+    Returns spatial bounds in the provided spatial reference
+    """
+
+    def _get_point(x, y):
+        x, y = (x, y) * transform
+        return {'x': x, 'y': y}
+
+    return {
+        'geo_ref_points': {
+            'ul': _get_point(0, 0),
+            'll': _get_point(0, shape[0]),
+            'ur': _get_point(shape[1], 0),
+            'lr': _get_point(shape[1], shape[0])
+        },
+        'spatial_reference': spatial_reference
+    }
+
+
+def process_dataset(input_path: Path, xml_file: Path):
+    """
+    Generates a metadata document for each tile provided,
+    requires a path to the input tile (hdf) and the
+    corresponding xml document describing the dataset.
+    """
+    band_info = get_band_info(input_path)
+    xml_md = parse_xml(xml_file)
+
+    md = {}
+    md['id'] = uuid.uuid5(MCD43A1_NS, xml_md.get('granule_id', input_path.name))
```

Comment:
I don't know anything about granule ids, but I presume they include processing time or proc version? (there's no example anywhere I can see in the diff or tests)

Fallbacks like `input_path.name` have been dangerous in the past (unexpected human usage of a script). Not sure about it.

Reply:
I'll remove the fallback :)